Source/ARTRealtime.m

```diff
@@ -540,6 +542,12 @@ - (void)updateWithErrorInfo:(nullable ARTErrorInfo *)errorInfo {
 - (void)handleNetworkIsReachable:(BOOL)reachable {
     if (reachable) {
         switch (_connection.state_nosync) {
+            case ARTRealtimeConnecting: {
+                if ([self shouldRestartPendingConnectionAttempt]) {
```

Comment:
The behaviour that you describe in the commit message should be described here too, I think — as you mentioned in Slack, it's a workaround and it should be brought to people's attention since it's not a behaviour described by the spec. Also I'd suggest creating an issue that describes the underlying reason for this workaround and what we need to do to remove it, and link to it here.

Reply:
https://github.com/ably/ably-cocoa/pull/1703/commits/08ba4c8fded2d94ec7efcecba95ccf979b9c80ca

Issue https://github.com/ably/ably-cocoa/issues/1713